### DISCUSSION TITLE
Create-select has no way to specify a default value?

### DISCUSSION THREAD

**Discussion by sabracrolleton** [rabbibotton/clog#121] [Q&A] [answered]
Looking at Tutorial 9 on forms, I do not see a way to pre-select (e.g. have a default value) in a select box. Am I missing something?

As a side note, I do not use cl-dbi, but reading the tutorials left me very puzzled on how I would access anything other than sqlite. It might be useful to have an example using accessing one of the other databases.

**Comment by rabbibotton** [comment, marked as answer]
> pre-select (e.g. have a default value) in a select box.

The main way to add options to a select box is to use the create-option method, to make it the selected object, just add :selected t
The convenience function add-select-option also has that option. add-select-option does not currently provide a way, but that would be a trivial enhancement. I will look in to it in the near future after my first iterations on clog-web-site are done and a few important features like alignment features in the builder.

> puzzled on how I would access anything other than sqlite

I will add some comments soon, as it is just a matter of how you initialize dbi.

mysql:

(defvar *connection*
  (dbi:connect :mysql
               :database-name "test"
               :username "nobody"
               :password "1234"))

postgres:

(defvar *connection*
  (dbi:connect :posgress
               :database-name "test"
               :username "nobody"
               :password "1234"))

sqlite:

 (defvar *connection*
  (dbi:connect :sqlite3
               :database-name "/home/gt/test.sqlite3"))

There is also a driver for Oracle as well and others could be added easily.

The github is at - https://github.com/fukamachi/cl-dbi

**Reply by sabracrolleton** [reply]
Thank you.
